plugin/pkg/admission/securitycontext/scdeny/admission.go

```diff
@@ -54,42 +54,52 @@ func (p *plugin) Admit(a admission.Attributes) (err error) {
 		return apierrors.NewBadRequest("Resource was marked with kind Pod but was unable to be converted")
 	}
 
-	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.SupplementalGroups != nil {
-		return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("SecurityContext.SupplementalGroups is forbidden"))
+	if err := checkPodSecurityContext(pod.Spec.SecurityContext); err != nil {
+		return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, err)
 	}
-	if pod.Spec.SecurityContext != nil {
-		if pod.Spec.SecurityContext.SELinuxOptions != nil {
-			return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("pod.Spec.SecurityContext.SELinuxOptions is forbidden"))
-		}
-		if pod.Spec.SecurityContext.RunAsUser != nil {
-			return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("pod.Spec.SecurityContext.RunAsUser is forbidden"))
+
+	for _, initContainer := range pod.Spec.InitContainers {
+		if err := checkSecurityContext(initContainer.SecurityContext); err != nil {
+			return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, err)
 		}
 	}
 
-	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.FSGroup != nil {
-		return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("SecurityContext.FSGroup is forbidden"))
+	for _, container := range pod.Spec.Containers {
+		if err := checkSecurityContext(container.SecurityContext); err != nil {
+			return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, err)
+		}
 	}
+	return nil
+}
 
-	for _, v := range pod.Spec.InitContainers {
-		if v.SecurityContext != nil {
-			if v.SecurityContext.SELinuxOptions != nil {
-				return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("SecurityContext.SELinuxOptions is forbidden"))
-			}
-			if v.SecurityContext.RunAsUser != nil {
-				return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("SecurityContext.RunAsUser is forbidden"))
-			}
-		}
+func checkPodSecurityContext(psc *api.PodSecurityContext) error {
```

Comment:
How about aggregating all possible errors? If users configure several SecurityContext options, they can immediately know all what are forbidden in their configuration. Otherwise, they will get one `forbidden` info and then change the configuration and then get another `forbidden` info.